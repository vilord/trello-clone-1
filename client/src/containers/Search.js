import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Input, Icon } from 'semantic-ui-react';

import { focusHeaderSearch, blurHeaderSearch } from '../actions/ui';

import './Search.css';

export class Search extends Component {
  render() {
    const { focusHeaderSearch, blurHeaderSearch, focused } = this.props;
    return (
      <Input
        className="Search"
        icon={
          <Icon
            name={focused ? 'close' : 'search'}
            flipped="horizontally"
            style={{
              color: focused ? '#999' : 'white',
            }}
          />
        }
        onFocus={focusHeaderSearch}
        onBlur={blurHeaderSearch}
      />
    );
  }
}

Search.propTypes = {
  focusHeaderSearch: PropTypes.func.isRequired,
  blurHeaderSearch: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  focused: state.ui.focus === 'search',
});

const mapDispatchToProps = dispatch => ({
  focusHeaderSearch: () => dispatch(focusHeaderSearch()),
  blurHeaderSearch: () => dispatch(blurHeaderSearch()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Search);
