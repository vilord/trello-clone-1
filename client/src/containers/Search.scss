.Search {
  display: flex;
  overflow: visible;
  align-items: center;

  margin: auto 4px;

  input {
    height: 30px;
    width: 180px;

    transition: width .1s ease-out !important;

    border: none !important;
    outline: none;
    background-color: #5395c0 !important;

    &:hover {
      background-color: #6ba4c9 !important;
    }

    &:focus {
      width: 360px;

      color: black !important;
      background-color: white !important;
    }
  }

  i {
    cursor: pointer !important;
    pointer-events: auto !important;

    opacity: 1 !important;
    color: white;

    &:focus {
      color: black !important;
    }
  }
}
