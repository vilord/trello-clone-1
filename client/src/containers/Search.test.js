import React from 'react';
import { shallow } from 'enzyme';

import { Search } from './Search';

describe('Search Component', () => {
  beforeAll(() => {
    console.error = err => {
      throw new Error(err);
    };
  });

  const minProps = {
    focusHeaderSearch: jest.fn(),
    blurHeaderSearch: jest.fn(),
  };

  let component;
  beforeEach(() => {
    component = shallow(<Search {...minProps} />);
  });

  it('renders', () => {
    expect(component).toHaveLength(1);
  });

  it('calls props.focusHeaderSearch on Input focus', () => {
    expect(minProps.focusHeaderSearch).not.toHaveBeenCalled();
    component.find('Input').simulate('focus');
    expect(minProps.focusHeaderSearch).toHaveBeenCalled();
  });

  it('calls props.blurHeaderSearch on Input blur', () => {
    component.find('Input').simulate('focus');
    expect(minProps.blurHeaderSearch).not.toHaveBeenCalled();
    component.find('Input').simulate('blur');
    expect(minProps.blurHeaderSearch).toHaveBeenCalled();
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
