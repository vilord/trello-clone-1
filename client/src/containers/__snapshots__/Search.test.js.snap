// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component matches snapshot 1`] = `
<Input
  className="Search"
  icon={
    <Icon
      as="i"
      flipped="horizontally"
      name="search"
      style={
        Object {
          "color": "white",
        }
      }
    />
  }
  onBlur={[Function]}
  onFocus={[Function]}
  type="text"
/>
`;
